### PR TITLE
[doc] Dealing with Lie group geometry 

### PR DESCRIPTION
**Pull request by olivier-stasse**
A small contribution in using SE(2) with pinocchion in C++.

### CODE REVIEW

**Comment by gabrielebndn** [comment]
Very nice!

File: doc/a-features/e-lie.md [hunk 1]
```diff
@@ -1 +1,77 @@
...
+## Using \\( SE(2) \\) with pinocchio in C++
+
+As a motivating example let us consider a mobile robot evolving in a plane \f$(\mathbb{R}^2 \times \mathbb{S}^1 \f$).
+![SE2MotivatingExample.svg](SE2MotivatingExample.svg)
+
+The robot starts at position \f$ pose_s=(x_s,y_s,\theta_s) \f$ and after a ridig motion
+\f$ \delta_u=(\delta x,\delta y,\delta \theta) \f$ 
+it is finishing 
+at position \f$ pose_g = (x_g,y_g,\theta_g)\f$.
+It is possible to instanciate the corresponding \\(SE(2)\\) objects using:
```
Comment (gabrielebndn) [inline doc/a-features/e-lie.md]: instanciate -> instantiate

File: doc/a-features/e-lie.md [hunk 2]
```diff
@@ -1 +1,77 @@
...
+It has various applications like representing the motion of a robot free flyer joint (typically the base of a mobile robot),
+or the motion of the robot links. The later is particularly useful for collision detection.
+It is also interesting to have general vector space over which a Lie algebra is defined. 
+
+## Using \\( SE(2) \\) with pinocchio in C++
+
+As a motivating example let us consider a mobile robot evolving in a plane \f$(\mathbb{R}^2 \times \mathbb{S}^1 \f$).
+![SE2MotivatingExample.svg](SE2MotivatingExample.svg)
+
+The robot starts at position \f$ pose_s=(x_s,y_s,\theta_s) \f$ and after a ridig motion
```
Comment (gabrielebndn) [inline doc/a-features/e-lie.md]: ridig -> rigid

File: doc/a-features/e-lie.md [hunk 3]
```diff
@@ -1 +1,77 @@
...
+\f$ \delta_u=(\delta x,\delta y,\delta \theta) \f$ 
+it is finishing 
+at position \f$ pose_g = (x_g,y_g,\theta_g)\f$.
+It is possible to instanciate the corresponding \\(SE(2)\\) objects using:
+
+\code
+  typedef double Scalar;
+  enum {Option = 0};
```
Comment (gabrielebndn) [inline doc/a-features/e-lie.md]: Option -> Options

File: doc/a-features/e-lie.md [hunk 4]
```diff
@@ -1 +1,77 @@
...
+  pose_s(2) = cos(M_PI/4.0); pose_s(3) = sin(M_PI/4.0);
+  pose_g(0) = 3.0; pose_g(1) = -1.0;
+  pose_g(2) = cos(-M_PI/2.0); pose_g(3) = sin(-M_PI/2.0);
+
+  aSE2.difference(pose_s,pose_g,delta_u);
+  std::cout << delta_u << std::endl;
+  pose_s = aSE2.random();
+  pose_g = aSE2.random();
+  aSE2.difference(pose_s,pose_g,delta_u);
+  std::cout << delta_u << std::endl;
```
Comment (gabrielebndn) [inline doc/a-features/e-lie.md]: I would remove these last four lines or put them somewhere else. Here, I find they are a bit confusing for what you say below about the result

File: examples/o-liegroup.cpp [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  aSE2.difference(pose_s,pose_g,delta_u);
+  std::cout << delta_u << std::endl;
+
+  aSE2.difference(pose_s,pose_g,delta_u);
+
+  SpecialEuclideanOperationTpl<2,Scalar,Options>::ConfigVector_t pose_check;
+  aSE2.integrate(pose_s,delta_u,pose_check);
+  std::cout << pose_check << std::endl;
+}
+
```
Comment (gabrielebndn) [inline examples/o-liegroup.cpp]: Nice! This could be included in the doc along with the other examples.

**Comment by olivier-stasse** [comment]
Thanks @gabrielebndn . The last commit takes into account your feedback.

**Comment by jcarpent** [comment]
@olivier-stasse Thanks for the fix.

**Pull request merged**